plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -2834,20 +2868,27 @@ private void testStatsBasedRepartitionData(boolean ctas)
     }
 
     private void testRepartitionData(Session session, String sourceRelation, boolean ctas, String partitioning, int expectedFiles)
+    {
+        testRepartitionData(session, sourceRelation, Optional.empty(), Optional.empty(), ctas, partitioning, expectedFiles);
+    }
+
+    protected void testRepartitionData(Session session, String sourceRelation, Optional<String> sourceOrderBy, Optional<Integer> limitSourceRows, boolean ctas, String partitioning, int expectedFiles)
     {
         String tableName = "repartition" +
                 "_" + sourceRelation.replaceAll("[^a-zA-Z0-9]", "") +
                 (ctas ? "ctas" : "insert") +
                 "_" + partitioning.replaceAll("[^a-zA-Z0-9]", "") +
                 "_" + randomTableSuffix();
 
-        long rowCount = (long) computeScalar(session, "SELECT count(*) FROM " + sourceRelation);
+        String orderByClause = sourceOrderBy.map(orderBy -> " ORDER BY " + orderBy).orElse("");
+        String limitClause = limitSourceRows.map(limit -> " LIMIT " + limit + " ").orElse("");
+        long rowCount = computeActual("SELECT * FROM " + sourceRelation + orderByClause + limitClause).getRowCount();
 
         if (ctas) {
             assertUpdate(
                     session,
                     "CREATE TABLE " + tableName + " WITH (partitioning = ARRAY[" + partitioning + "]) " +
-                            "AS SELECT * FROM " + sourceRelation,
+                            "AS SELECT * FROM " + sourceRelation + orderByClause + limitClause,
```

Comment:
No need for `Optional<String> sourceOrderBy, Optional<Integer> limitSourceRows`.
the caller can embed this directly in `sourceRelation`
i.e. let the caller provide the `sourceQuery` (either "TABLE table_name", or "SELECT * FROM ... ORDER BY LIMIT")